api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -84,6 +84,14 @@ message RouteAction {
     // assigned to each cluster.
     WeightedCluster weighted_clusters = 2;
   }
+
+  // Optional endpoint metadata match criteria used by the subset load balancer. Only endpoints in
+  // the upstream cluster with metadata matching what is set in this field will be considered.
+  // Note that this will be merged with what's provided in :ref: `WeightedCluster.MetadataMatch
+  // <envoy_api_field_config.filter.network.thrift_proxy.v2alpha1.WeightedCluster.ClusterWeight.metadata_match>`,
+  // with values there taking precedence. Keys and values should be provided under the "envoy.lb"
+  // metadata key.
+  envoy.api.v2.core.Metadata metadata_match = 3;
```

Comment:
a few questions:

a) are we planning on supporting something like https://github.com/envoyproxy/envoy/commit/2d2506761a3c9d03594619f7c6cc8596ddb59950 so that we can write filters that sets dynamic metadata based on a thrift header?

b) do thrift endpoints support mutable metadata like we do for HTTP (cfr https://github.com/envoyproxy/envoy/commit/810ec2ec52fc9ccb305b41165cb7b8062fe59248)?

Reply:
1) i don't know of any concrete plans to do so since we don't keep track of `RequestInfo` in the thrift connection manager right now
2) it's all the same cluster/upstream management so it should just work